### DISCUSSION TITLE
Ability to override pre/postfix for auth, ability to change what gets replaced

### DISCUSSION THREAD

**Discussion by hybridherbst** [n4bb12/verdaccio-github-oauth-ui#161] [Ideas]
## Feature Request

Currently, the 2nd and 3rd lines in the Registry Info will be replaced by a [hardcoded set of strings](https://github.com/n4bb12/verdaccio-github-oauth-ui/blob/5455fdb471417f39fb37a91d0334a2bd44b48ac1/src/client/plugin/usage-info.ts#L20). It would be nice if this was configurable (e.g. via config.yaml in Verdaccio), for two reasons:
- we're using a different package manager (Unity Package Manager) and want to adjust the text there
- currently the text becomes very long when a package is selected (so all commands look the same).

#### Is your feature request related to a problem? Please describe.
See above

#### Describe the solution you'd like
1. Ability to use Verdaccio's config.yaml to specify a string, e.g. "myprefix {{auth}} mypostfix"
2. Ability to specify if the always-auth command is also replaced
3. Ability to specify in the config under which conditions this is replaced (e.g. we only want that "Unity Package Manager" tab actually)

#### Additional context
See very long text here
![image](https://user-images.githubusercontent.com/2693840/142093301-514241f0-6659-4b60-aad3-6c908f605c89.png)

**Comment by n4bb12** [comment]
I think the idea makes sense.

Verdaccio by default shows tabs for yarn and pnpm. Those commands are currently also not adjusted by the plugin.

Making this configurable sounds a little complicated to me. Especially the conditional stuff.

I wonder why Verdaccio shows 3 copy buttons instead of one. I could imagine replacing this with a single copy button and a single multiline text. Then this text could be configured for each package manager and decisions like whether to replace the always-auth command would be implicit since you would just configure whatever you want to see.

What are your thoughts and do you have a suggestion for how this could look like in yaml?

**Comment by n4bb12** [comment]
@hybridherbst any thoughts?

**Comment by hybridherbst** [comment]
Still playing around in our fork, haven't come to a great solution yet, and some trouble understanding Verdaccio config and how to bubble it through an app it seems 😅
But basically, what I was trying is to
a) specify in the Verdaccio theme which tabs are rendered
b) specify in this plugin how and where the auth code is pasted, e.g. different regex

I'll update this with more info once I have a result I like - please feel free to turn this issue into a discussion in the meantime :)
